app/scripts/controllers/detect-tokens.js

```diff
@@ -192,6 +184,9 @@ export default class DetectTokensController extends StaticIntervalPollingControl
         result = await this.assetsContractController.getBalancesInSingleCall(
           addressAgainstWhichToDetect,
           tokensSlice,
+          this.network.findNetworkClientIdByChainId(
+            chainIdAgainstWhichToDetect,
+          ),
```

Comment:
This feels like maybe the sketchiest one.  We weren't passing an explicit `NetworkClientId` here, which means  `AssetsContractController` falls back to the chain in its state.  But that controller's `onNetworkDidChange` has not always fired yet, so its still on the old chain, and we detect the new token addresses on the old chain.

This was the only way I found to go from chain id -> `NetworkClientId` since that's what `getBalancesInSingleCall` accepts.

Reply:
Hey Brian, yeah so this is the best option available for right now. You've correctly identified an issue with this flow. We have implemented an alternative polling solution that we will be leveraging shortly but you've highlighted one piece that we're missing. @shanejonas we will need to pass along the networkClientId in the `_executePoll` method.

Reply:
created a ticket to do that cleanup: https://app.zenhub.com/workspaces/devex-63bee08a4e3b9d001108416e/issues/gh/metamask/metamask-planning/2097

Reply:
https://github.com/MetaMask/metamask-extension/pull/22898